packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -0,0 +1,1102 @@
+                slashed = -1 * int256(voterStake.cumulativeStaked * requestSlashingTrackers[i].wrongVoteSlashPerToken);
+
+                // The voter voted correctly.Receive a pro-rate share of the other voters slashed amounts as a reward.
+            else {
+                uint256 roundId = rounds[priceRequestIds[i].roundId].snapshotId;
+                uint256 totalStaked = votingToken.balanceOfAt(address(this), roundId);
+                slashed = int256(
+                    ((voterStake.cumulativeStaked * 1e18) / totalStaked) * requestSlashingTrackers[i].totalSlashed
+                );
+            }
+            if (slashed + int256(voterStake.cumulativeStaked) > 0)
+                voterStake.cumulativeStaked = uint256(int256(voterStake.cumulativeStaked) + slashed);
+            else voterStake.cumulativeStaked = 0;
+        }
+    }
+
+    function _updateCumulativeSlashingTrackers() internal {
+        // Note the method below can hit a gas limit of there are a LOT of requests from the last time this was run.
+        // A future version of this should bound how many requests to look at per call to avoid gas limit issues.
+        for (uint256 i = lastRequestIndexConsidered; i < priceRequestIds.length; i++) {
+            PriceRequest storage priceRequest = priceRequests[priceRequestIds[i].requestId];
+            VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
+            uint256 roundId = rounds[priceRequestIds[i].roundId].snapshotId;
+
+            uint256 totalStaked = votingToken.balanceOfAt(address(this), roundId);
+            uint256 totalVotes = voteInstance.resultComputation.totalVotes.rawValue;
+            uint256 totalCorrectVotes = voteInstance.resultComputation.getTotalCorrectlyVotedTokens().rawValue;
+
+            uint256 wrongVoteSlashPerToken = calcWrongVoteSlashPerToken(totalStaked, totalVotes, totalCorrectVotes);
+            uint256 noVoteSlashPerToken = calcNoVoteSlashPerToken(totalStaked, totalVotes, totalCorrectVotes);
+
+            uint256 totalSlashed = ((noVoteSlashPerToken + wrongVoteSlashPerToken) * totalStaked) / 1e18;
+            requestSlashingTrackers.push(SlashingTracker(wrongVoteSlashPerToken, noVoteSlashPerToken, totalSlashed));
+        }
+        lastRequestIndexConsidered = priceRequestIds.length;
+    }
+
+    // Note the functions below should be pulled out into an external library that is paramaterizable by each of the
+    // props passed in is uppdatable via governance.
+
+    function calcWrongVoteSlashPerToken(
+        uint256 totalStaked,
+        uint256 totalVotes,
+        uint256 totalCorrectVotes
+    ) public pure returns (uint256) {
+        return 1600000000000000;
+    }
+
+    function calcNoVoteSlashPerToken(
+        uint256 totalStaked,
+        uint256 totalVotes,
+        uint256 totalCorrectVotes
+    ) public pure returns (uint256) {
+        return 1600000000000000;
+    }
```

Comment:
this is the MVP of the slashing logic. a subsequent PR will update this and add unit tests